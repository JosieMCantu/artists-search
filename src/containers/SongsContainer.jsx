import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import SongList from '../components/songs/SongList';
import { getSongsById } from '../services/apiUtils';

export const useSongs = () => {
  const [loading, setLoading] = useState(true);
  const [songs, setSongs] = useState([]);
  const { artistName, id } = useParams();

  useEffect(() => {
    getSongsById(id)
      .then((songs) => setSongs(songs))
      .finally(() => setLoading(false));
  }, []);
  return [loading, songs, artistName];
};

const SongsContainer = () => {
  const [loading, songs, artistName] = useSongs();

  return (
    <div>
      {loading ? (
        <h1>LOADING...</h1>
      ) : (
        <SongList songs={songs} artistName={artistName} />
      )}
    </div>
  );
};
export default SongsContainer;
