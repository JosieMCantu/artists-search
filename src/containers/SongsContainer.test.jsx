/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render, screen, waitFor} from '@testing-library/react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import SongsContainer from './SongsContainer';
import { MemoryRouter } from 'react-router';
import mockData from '../../fixtures/mock-data-song.json'

const server = setupServer(
    rest.get(`http://musicbrainz.org/ws/2/recording?release=59211ea4-ffd2-4ad9-9a4e-941d3148024a&fmt=json`, (req, res, ctx) => {
        return res(ctx.json(mockData));
    })
);

describe('SongsContainer', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    
    it('renders a loading element', async () => {
        render(<MemoryRouter><SongsContainer/></MemoryRouter>);
        const loading = await screen.getByText('LOADING...');
         return waitFor(()=> {
          expect(loading).toMatchSnapshot();  
        });
    });

    it('renders a list of releases by an artists', async ()=> {
        render(<MemoryRouter><SongsContainer/></MemoryRouter>);
        const ul = await screen.findByRole('list', { name: 'songs'});
        return waitFor(()=> {
          expect(ul).toMatchSnapshot();  
        });
        
    });
});
