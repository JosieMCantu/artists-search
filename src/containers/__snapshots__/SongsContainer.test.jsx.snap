// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongsContainer renders a list of releases by an artists 1`] = `
<ul
  aria-label="songs"
>
  <li>
    <a
      href="/song/undefined/h³æ"
    >
      <div>
        <p>
          h³æ
        </p>
      </div>
    </a>
  </li>
  <li>
    <a
      href="/song/undefined/æ³o"
    >
      <div>
        <p>
          æ³o
        </p>
      </div>
    </a>
  </li>
</ul>
`;

exports[`SongsContainer renders a loading element 1`] = `
<h1>
  LOADING...
</h1>
`;
